iroha/src/sumeragi.rs

```diff
@@ -529,24 +537,27 @@ impl Sumeragi {
 
     #[log]
     async fn handle_block_committed(&mut self, mut block: SignedBlock) -> Result<(), String> {
-        //TODO: check if the block is the same as pending (set b has to be aware of the block)
-        let verified_signatures = block.verified_signatures();
-        let valid_signatures = self.network_topology.filter_signatures_by_roles(
-            &[Role::ValidatingPeer, Role::Leader, Role::ProxyTail],
-            &verified_signatures,
-        );
-        let proxy_tail_signatures = self
-            .network_topology
-            .filter_signatures_by_roles(&[Role::ProxyTail], &verified_signatures);
-        if valid_signatures.len() >= self.network_topology.min_votes_for_commit()
-            && proxy_tail_signatures.len() == 1
-        {
-            block.signatures.clear();
-            block.signatures.append(&valid_signatures);
-            let block = block.validate(&*self.world_state_view.read().await)?;
-            let hash = block.hash();
-            self.blocks_sender.write().await.send(block).await;
-            self.next_round(hash).await;
+        let voting_block = self.voting_block.write().await.clone();
```

Comment:
unnecessary local variable

Reply:
I wanted to to skip it at first but it turns out to be necessary.
` if let Some(voting_block) = self.voting_block.write().await.clone() {
    |                                     -------------------------------
    |                                     |
    |                                     immutable borrow occurs here
    |                                     a temporary with access to the immutable borrow is created here ...
...
550 |                 self.next_round(hash).await;
    |                 ^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
`